pkg/route/validation/validation_test.go

```diff
@@ -1200,6 +1687,119 @@ func TestValidateRouteUpdate(t *testing.T) {
 	}
 }
 
+// This unit test tests the regex for validating various values which a user can pass.
+func TestHeaderValues(t *testing.T) {
+	// The syntax is http-request set-header <name> <fmt> [ { if | unless } <condition> ]. As per  As per this http://cbonte.github.io/haproxy-dconv/2.6/configuration.html#4.2-http-request%20add-header
+	// the format  follows the log-format rules (see Custom Log Format in section 8.2.4) i.e http://cbonte.github.io/haproxy-dconv/2.6/configuration.html#8.2.4
+	// The usage of it shall be as per section 8.2.6
+	// However for dynamic values, req.hdr, res.hdr, ssl_c_der are the allowed fetchers and lower, base64 are the allowed converters.
+	type HeaderValueTest struct {
+		description string
+		validInput  bool
+		input       string
+	}
+
+	tests := []HeaderValueTest{
+		{description: "empty value", input: ``, validInput: false},
+		{description: "single character", input: `a`, validInput: true},
+		{description: "multiple characters", input: `abc`, validInput: true},
+		{description: "multiple words without escaped space", input: `abc def`, validInput: true},
+		{description: "multiple words with escaped space", input: `abc\ def`, validInput: true},
+		{description: "multiple words each word quoted", input: `"abc"\ "def"`, validInput: true},
+		{description: "multiple words each word quoted and with an embedded space", input: `"abc "\ "def "`, validInput: true},
+		{description: "multiple words each word one double quoted and other single quoted and with an embedded space", input: `"abc "\ 'def '`, validInput: true},
+		{description: "single % character", input: `%`, validInput: false},
+		{description: "escaped % character", input: `%%`, validInput: true},
+		{description: "escaped % and only a % character", input: `%%%`, validInput: false},
+		{description: "two literal % characters", input: `%%%%`, validInput: true},
+		{description: "zero percent", input: `%%%%%%0`, validInput: true},
+		{description: "escaped expression", input: `%%[XYZ.hdr(Host)]\ %[XYZ.hdr(Host)]`, validInput: true},
+		{description: "simple empty expression", input: `%[]`, validInput: false},
+		{description: "nested empty expressions", input: `%[%[]]`, validInput: false},
+		{description: "empty quoted value", input: `%{+Q}`, validInput: false},
+		{description: "quoted value", input: `%{+Q}foo`, validInput: false},
```

Comment:
Why is this invalid?

Reply:
There is no [] after }

Reply:
Can you add one test using `{+Q}` that is valid, for reference?

Reply:
Please add one test using `%{+Q}` that is valid.

Reply:
added

Reply:
It is not here.

Reply:
https://github.com/openshift/library-go/pull/1449/files#diff-51cbcb9e560e512d7dba0cd93dab386055814a27e2c7903ad93bce92c58fd18bR2081